Orange/data/io.py

```diff
@@ -156,20 +177,17 @@ def valuemap_index(val):
             # Never use existing for un-named variables
             var = coltype(new_var_name, **coltype_kwargs)
 
-        # Reorder discrete values to match existing variable
         if var.is_discrete and not var.ordered:
-            new_order, old_order = var.values, coltype_kwargs.get('values',
-                                                                  var.values)
-            if new_order != old_order:
-                offset = len(new_order)
-                column = values if data.ndim > 1 else data
-                column += offset
-                for _, val in enumerate(var.values):
-                    try:
-                        oldval = old_order.index(val)
-                    except ValueError:
-                        continue
-                    bn.replace(column, offset + oldval, new_order.index(val))
+            # Reorder discrete values to match existing variable
+            reorder_values()
+
+    if isinstance(var, ContinuousVariable) and var.adjust_decimals:
+        ndecimals = get_number_of_decimals()
+        if var.adjust_decimals == 2 or ndecimals > var.number_of_decimals:
```

Comment:
I do not understand what `var.adjust_decimals == 2` means

Reply:
See here: https://github.com/biolab/orange3/blob/master/Orange/data/variable.py#L439

Reply:
It would be nice to have named constants instead of magic values (for instance: 0=NEVER, 1=MORE_PRECISE, 2=ALWAYS).
A comment why this is needed here would be nice as well, as the documentation on the variable states that this behaviour is what ContinuousVariable does by itself.

The code also differs a bit from the documented behaviour of the ContinuousVariable: adjust_decimals is supposed to be set to 1 after number_of_decimals is set for the first time.

Reply:
I added a comment and changed the code so that it sets `adjust_decimals` to 1. I removed a redundant part where this couldn't be done.

I agree about the constants, but not in this PR. In particular because `ContinuousVariable.to_val` no longer adjusts the number of decimals, so other fixes are needed there. I'll open an issue to discuss this.